src/main/java/com/google/sps/servlets/BooksServlet.java

```diff
@@ -0,0 +1,148 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import com.google.gson.Gson;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.sps.data.Book;
+import java.net.URL;
+import java.net.HttpURLConnection;
+import java.util.Scanner;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonElement;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.net.URLEncoder;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Servlet for generating books from keywords.
+ */
+@WebServlet("/books")
+public class BooksServlet extends HttpServlet {
+
+  private final int NUM_BOOKS_PER_KEYWORD = 5;
+  private String[] keywords;
+  private LinkedHashMap<String, List<Book>> booksMap = new LinkedHashMap<>();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    keywords = request.getParameterValues("key");
+    for (String keyword : keywords) {
```

Comment:
[Optional] Use a lambda expression to evaluate very value without a for loop.
https://stackoverflow.com/questions/7221833/how-can-i-call-a-method-on-each-element-of-a-list/20684006#20684006